### PR TITLE
flathub submission for com.itextpdf.rups

### PR DESCRIPTION
**Pull request by nbenitez**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ ] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by nbenitez** [comment]
About having no contacted upstream, I must say this project is difficult to contact as they don't have a public issue tracker, see:\
https://itextpdf.com/en/start-contributing#issue

they mirror the code on github, but their github page does not have the issue tracker enabled:\
https://github.com/itext/i7j-rups

I don't own nor I'm employed by `itextpdf.com`. I submitted app ID as `com.itextpdf.rups` but I can resubmit as `com.github.itext.i7j-rups` if that is better with flathub guidelines.

File: com.github.itext.i7j_rups.json [hunk 1]
```diff
@@ -0,0 +1,120 @@
+{
+    "id": "com.github.itext.i7j_rups",
```
Comment (barthalion) [inline com.github.itext.i7j_rups.json:2]:
```suggestion
    "id": "com.itextpdf.RUPS",
```

File: com.github.itext.i7j_rups.json [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    "finish-args": [
+        "--env=PATH=/app/jre/bin:/app/bin:/usr/bin",
+        "--env=JAVA_HOME=/app/jre",
+        "--env=JAVA=/app/jre/bin/java",
+        "--share=ipc",
+        "--socket=x11",
+        "--socket=pulseaudio",
+        "--filesystem=xdg-documents",
+        "--filesystem=xdg-download",
+        "--filesystem=xdg-pictures",
```
Comment (barthalion) [inline com.github.itext.i7j_rups.json:17]:
```suggestion
        "--filesystem=xdg-documents",
        "--filesystem=xdg-download",
```

I'm not sure if it shouldn't be `home` though.

File: com.github.itext.i7j_rups.json [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+            "build-commands": [
+                "/usr/lib/sdk/openjdk11/install.sh",
+                "install -d /app/rups/",
+                "mv rups/* /app/rups/",
+                "install -Dm755 rups.sh /app/bin/rups"
+            ],
+            "post-install": [
+                "desktop-file-install --dir=/app/share/applications com.github.itext.i7j_rups.desktop",
+                "install -Dm644 com.github.itext.i7j_rups.metainfo.xml /app/share/metainfo/com.github.itext.i7j_rups.metainfo.xml",
+                "for i in 32 48 64 96 128 256 512; do
```
Comment (barthalion) [inline com.github.itext.i7j_rups.json:88]: Just 512x512 is enough.

**Comment by barthalion** [comment]
Hi @Snipx, I noticed you are a recent contributor to RUPS repo. Is there a good place we could contact iText about distributing your app on [Flathub](https://flathub.org)?

**Comment by nbenitez** [comment]
I just sent an email to community@itextpdf.com , hopefully we'll get an answer.

**Comment by hfiguiere** [comment]
bot, build com.github.itext.i7j_rups

**Comment by barthalion** [comment]
@nbenitez Check your e-mail, I've added you to a thread started by Alexey. (Thanks!)

**Pull request closed without merging**